sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -641,14 +641,19 @@ object SQLConf {
     .checkValue(_ > 0, "The value of spark.sql.leafNodeDefaultParallelism must be positive.")
     .createOptional
 
+  // Lazily get the number of cores to make sure SparkContext created first.
+  private lazy val defaultShufflePartition: Option[Int] = SparkContext.getActive.flatMap { sc =>
+    if (sc.master.startsWith("local")) Some(SparkContext.numDriverCores(sc.master)) else None
```

Comment:
This looks like it will break the `local-cluster` mode. With this change, `SHUFFLE_PARTITIONS` seems to be configured to 0 in `local-cluster` mode?

Reply:
oh yeah